Xamarin.Forms.Core/Picker.cs

```diff
@@ -58,5 +78,39 @@ void OnItemsCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
 		{
 			SelectedIndex = SelectedIndex.Clamp(-1, Items.Count - 1);
 		}
+
+		static void OnItemsSourcePropertyChanged (BindableObject bindable, object value, object newValue)
+		{
+			var picker = (Picker)bindable;
+			var notifyCollection = newValue as INotifyCollectionChanged;
+			if (notifyCollection != null) {
+				notifyCollection.CollectionChanged += (sender, args) => {
+					if (args.NewItems != null)
+						foreach (var newItem in args.NewItems)
+							picker.Items.Add ((newItem ?? "").ToString ());
+
+					if (args.OldItems != null)
+						foreach (var oldItem in args.OldItems)
+							picker.Items.Remove ((oldItem ?? "").ToString ());
+				};
+			}
```

Comment:
As elements are object in the ItemsSource, you want to listen to INPC on each of them, and update the list accordingly